rust/gui-client/src-tauri/src/client/logging.rs

```diff
@@ -123,23 +124,65 @@ pub(crate) async fn clear_logs_inner() -> Result<()> {
                 return Err(error.into());
             }
         };
+        let mut paths = vec![];
         while let Some(entry) = dir.next_entry().await? {
-            let path = entry.path();
-            if let Err(error) = tokio::fs::remove_file(&path).await {
-                tracing::error!(
-                    ?error,
-                    path = path.display().to_string(),
-                    "Error while removing log file"
-                );
-                // We'll return the most recent error, it loses some information but it's better than nothing.
-                result = Err(error);
+            paths.push(entry.path());
+        }
+
+        let to_delete = choose_logs_to_delete(&paths);
+        for path in &to_delete {
+            if let Err(e) = tokio::fs::remove_file(path).await {
+                result = Err(e);
             }
         }
     }
 
     Ok(result?)
 }
 
+fn choose_logs_to_delete(paths: &[PathBuf]) -> Vec<&Path> {
```

Comment:
Perhaps the following algorithm would be easier to understand:

- Given a file extension and a path
- List all files in a directory into a `Vec` (also filtering by `connlib.`)
- Sort using `.sort`
- Pop-off the last one
- Return list

Run call this fn twice, once for `.log` and once for `.jsonl`

Reply:
I know it's premature optimization, but I was thinking this runs in linear time whereas a sort will be super-linear